src/main/java/org/openmrs/addonindex/service/ElasticSearchIndex.java

```diff
@@ -156,7 +156,18 @@ public AddOnInfoAndVersions getByUid(String uid) throws IOException {
 		return client.execute(new Get.Builder(AddOnInfoAndVersions.ES_INDEX, uid).build())
 				.getSourceAsObject(AddOnInfoAndVersions.class);
 	}
-	
+
+	@Override
+	public AddOnInfoAndVersions getByModulePackage(String modulePackage) throws IOException {
+		return client.execute(new Search.Builder(new SearchSourceBuilder()
+				.size(1)
+				.query(QueryBuilders.matchQuery("modulePackage", modulePackage)).toString())
+				.addIndex(AddOnInfoAndVersions.ES_INDEX)
+				.build()).getHits(AddOnInfoAndVersions.class).stream()
+				.map(sr -> sr.source)
+				.collect(Collectors.toList()).get(0);
```

Comment:
At first glance I expect this will throw an ArrayIndexOutOfBoundsException if there are no hits.
Can you verify via the ManualTest class, and fix this. (I.e. it should return null, not throw an exception, if you give an incorrect modulePackage.)